doc/modules/preprocessing.rst

```diff
@@ -426,6 +426,47 @@ features, one has to explicitly set ``n_values``. For example,
 See :ref:`dict_feature_extraction` for categorical features that are represented
 as a dict, not as integers.
 
+.. _discretization:
+
+Discretization
+==============
+
+`Discretization <https://en.wikipedia.org/wiki/Discretization_of_continuous_features>`_
+provides a way to quantize continuous features into discrete values. Certain
+datasets with many continuous features may benefit from discretization, because
+discretization can transform the dataset of continuous attributes to one with
+only nominal attributes.
+
+Scikit-learn provides a mechanism way for users to discretize a feature into
+``k`` equal length bins using a :class:`KBinsDiscretizer`::
+
+  >>> X = np.array([[ -3., 5., 15 ],
+  ...               [  0., 6., 14 ],
+  ...               [  6., 3., 11 ]])
+  >>> est = preprocessing.KBinsDiscretizer(n_bins=[3, 3, 2]).fit(X)
+
+In this example, the "bin edges" (boundaries for the discretization criteria)
+for each of the three features are as follows:
+
+ - ``0`` and ``3``
+ - ``4`` and ``5``
+ - ``13``
+
+Based on these bin edges, we transform `X` as follows::
+
+  >>> est.transform(X)                      # doctest: +NORMALIZE_WHITESPACE
```

Comment:
Looking at Travis, I'm actually getting this error for the Python 3.6 build:

```
======================================================================
FAIL: Doctest: preprocessing.rst
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/testenv/lib/python3.6/doctest.py", line 2199, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for preprocessing.rst
  File "/home/travis/build/scikit-learn/scikit-learn/doc/modules/preprocessing.rst", line 0
----------------------------------------------------------------------
File "/home/travis/build/scikit-learn/scikit-learn/doc/modules/preprocessing.rst", line 457, in preprocessing.rst
Failed example:
    est.transform(X)                      # doctest: +NORMALIZE_WHITESPACE
Expected:
    array([[ 0., 2., 1.],
           [ 1., 2., 1.],
           [ 2., 0., 0.]])
Got:
    array([[-0.,  2.,  1.],
           [ 1.,  2.,  1.],
           [ 2., -0., -0.]])
```

Besides throwing in `# doctest: +SKIP`, do you have any suggestion how to avoid this doctest error?